packages/core/hooks/useMessageExternalSite.tsx

```diff
@@ -0,0 +1,110 @@
+import React from "react";
+
+type MessageExternalSiteContextType = {
+    send: (url: string, message: NonNullable<unknown>) => void;
+    setOnReceive: (onReceive: ((message: unknown) => void) | undefined) => void;
+};
+
+const noop = () => undefined;
+const MessageExternalSiteContext = React.createContext<MessageExternalSiteContextType>({
+    send: noop,
+    setOnReceive: noop,
+});
+
+/**
+ * A provider component for the `useMessageExternalSite` hook, which lazily creates and manages an
+ * invisible `iframe` when we want to post a message to an external site.
+ *
+ * This exists separately from the hook so that the `iframe` can be placed in a stable location in
+ * the document, allowing the hook to be used in components that unmount and re-mount often without
+ * reloading the `iframe` over and over.
+ */
+const MessageExternalSiteProvider: React.FC = ({ children }) => {
+    const [iFrameUrl, setIFrameUrl] = React.useState<string | null>(null);
+    const onReceiveRef = React.useRef<((message: unknown) => void) | undefined>(undefined);
+    const pendingMessageRef = React.useRef<NonNullable<unknown> | null>(null);
```

Comment:
I don't think I've seen `NonNullable` used like this, whats the benefit of providing it here?